### PR TITLE
[converter] Harden mixed-source SymInt lowerings under dynamic shapes

### PR DESCRIPTION
**Pull request by gokulkrishna98**
## Description
  Six small fixes in `_aten_to_core.py` / `_utils.py`, all surfacing from the same root cause: FX graphs that mix SymInt-derived Values (varying ranks / element types) with plain-int constants flow into coreai ops whose verifiers require uniform operands.

  1. **Register bare `aten.pow`** in `_op_map` and resolver — some `torch.export` rewrites leave `aten.pow` as the bare overload-packet target.
  2. **Same for bare `aten.round`.**
  3. **`upsample_build_output_shape_dynamic`** — normalise each `(out_h, out_w)` to rank-1 `si32` before the output-shape concat.
  4. **`get_operand` mixed-list path** — same normalisation for the dim-vector concat used by `view` / `expand` / `reshape` / `repeat`.
  5. **`replace_cat`** — when one input has a dynamic non-concat axis and a sibling has a known static size for the same axis, reshape the dynamic side to the static size before concat. Conflicting static sizes are left for the verifier (real shape mismatch).
  6. **`replace_arange_start_step`** — unify `start` / `end` / `step` to the FX-node output dtype before `coreai.range_`, mirroring aten's internal promotion.

  Fixes 3 and 4 share a new `to_rank1_int32(v)` helper.
  
## Testing
- python unit test
- enables conversion of dynamic config of FLMNORMALS

### CODE REVIEW

File: coreai_torch/_aten_to_core.py [hunk 1]
```diff
@@ -944,6 +948,36 @@ def replace_cat(values_map: dict[str, Value], node: fx.Node, loc: Location) -> V
 
     rank = inputs[0].type.rank
     dim = dim + rank if dim < 0 else dim
+
+    # coreai.concat requires all non-concat dims to be provably equal across
```
Comment (TobyRoseman) [inline coreai_torch/_aten_to_core.py:952]: It's not clear to me what a "sibling" is here.

Do we know for sure the dialect verifier will reject it? If so, why don't we just reject it?

Comment (gokulkrishna98) [reply coreai_torch/_aten_to_core.py:952]: Sibling here is the other inputs to the same concat operation. 

tldr of this change:
For a given non-concat axis, if one input has a dynamic size there but another input has a known static size, concat semantics require they're equal, so we promote the dynamic side to that static size.

> Do we know for sure the dialect verifier will reject it? 

Yes that is the error raised by the FLMNormals model, there is a radar for it.

> If so, why don't we just reject it?

I am not sure, what you mean here. If we reject it, FLMNormals model cannot be converted.

**Review by cymbalrush — APPROVED** [review]
